File: 167 Two Sum II - Input array is sorted/twoSumII.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0, r = numbers.length - 1;
        
        while(l < r){
            if(numbers[l] + numbers[r] == target){
                return new int[] {l+1, r+1}; //Return the results as an array
            }else if(numbers[l] + numbers[r] < target)
                l++;
            else
                r--;
        }
        return new int[] {};
    }
}

File: 167 Two Sum II - Input array is sorted/twoSumII.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {

        int l = 0, r = numbers.size() - 1;
        while(l < r){
            if(numbers[l] + numbers[r] == target){
                int res[2] = {l+1, r+1};
                return vector<int>(res, res+2);
            }else if(numbers[l] + numbers[r] < target)
                l++;
            else
                r--;
        }
        throw invalid_argument("The input has no solution.");
    }
};